nncf/pruning/filter_pruning/algo.py

```diff
@@ -281,62 +287,129 @@ def _find_layerwise_pruning_rate(self, target_flops_pruning_rate):
+        return pruned_param_count / full_param_count
+
+    @property
+    def current_groupwise_pruning_rate(self) -> Dict[int, float]:
+        """
+        Return the dict of layer group id's and corresponding current groupwise
+        pruning rates in the model
+        """
+        groupwise_pruning_rate_dict = {}
+        for group in self.pruned_module_groups_info.get_all_clusters():
+            groupwise_pruning_rate_dict[group.id] = self.pruning_rate_for_mask(group.nodes[0])
+        return groupwise_pruning_rate_dict
+
+    def _set_binary_masks_for_pruned_modules_groupwise(self,
+                                                       pruning_rate: Union[float, Dict[int, float]]) -> None:
+        """
+        Set the binary mask values according to groupwise pruning rates.
+        If pruning_rate is a float, set the pruning rates uniformly across groups.
+        If pruning_rate is a dict, set specific pruning rates corresponding to each group.
+        """
+        nncf_logger.debug("Updating binary masks for pruned modules.")
+        groupwise_pruning_rates_set = isinstance(pruning_rate, dict)
+
+        for group in self.pruned_module_groups_info.get_all_clusters():
+            group_pruning_rate = pruning_rate[group.id] if groupwise_pruning_rates_set \
+                else pruning_rate
+
+            filters_num = torch.tensor([get_filters_num(minfo.module) for minfo in group.nodes])
+            assert torch.all(filters_num == filters_num[0])
+            device = group.nodes[0].module.weight.device
+
+            cumulative_filters_importance = torch.zeros(filters_num[0]).to(device)
+            # 1. Calculate cumulative importance for all filters in group
+            for minfo in group.nodes:
+                filters_importance = self.filter_importance(minfo.module.weight,
+                                                            minfo.module.target_weight_dim_for_compression)
+                cumulative_filters_importance += filters_importance
+
+            # 2. Calculate threshold
+            num_of_sparse_elems = get_rounded_pruned_element_number(cumulative_filters_importance.size(0),
+                                                                    group_pruning_rate)
+            threshold = sorted(cumulative_filters_importance)[min(num_of_sparse_elems, filters_num[0] - 1)]
+            mask = calculate_binary_mask(cumulative_filters_importance, threshold)
+
+            # 3. Set binary masks for filter
+            for minfo in group.nodes:
+                pruning_module = minfo.operand
+                pruning_module.binary_filter_pruning_mask = mask
 
         # Calculate actual flops with new masks
         self.current_flops = self._calculate_flops_pruned_model_by_masks()
 
-    def _set_binary_masks_for_all_pruned_modules(self, pruning_rate):
+    def _set_binary_masks_for_pruned_modules_globally(self, pruning_rate: float) -> None:
```

Comment:
Why `_set_binary_masks_for_pruned_modules_globally` is still needed since `_set_binary_masks_for_pruned_modules_groupwise` can work with float `pruning_rate`?